### ISSUE TITLE
toODataString translate local date object into wrong UTC time

### ISSUE THREAD

**Issue by hung-doan** [telerik/kendo-angular#1730]
## I'm submitting a...

* Bug report 

## Current behavior

```typescript
let state = {
      filter: [
        {
          filters: [
            {
              field: "createDate",
              operator: "eq",
              value: new Date('Mon Aug 13 2018 00:00:00 GMT+0700 (Indochina Time)')
            }
          ],
          logic: "and"
        }
      ]
    };

let queryString = `${toODataString(state)}&$count=true`;
```
Will Result: 

```
/api/product?$filter=createDate%20ge%202018-08-13T00:00:00.000Z&$skip=0&$top=5&$count=true
```

You can see that It translate 'Mon Aug 13 2018 00:00:00 GMT+0700 (Indochina Time)' in to 2018-08-13T00:00:00.000Z (UTC), which is wrong

## Expected behavior

I expect that ` 2018 00:00:00 GMT+0700` will be translated into ` 2018-08-12T17:00:00.000Z (UTC)` instead of `2018-08-13T00:00:00.000Z (UTC)`

## Environment

Package versions:

```
+-- @angular-devkit/build-angular@0.6.8
+-- @angular/animations@6.1.2
+-- @angular/cli@6.0.8
+-- @angular/common@6.1.2
+-- @angular/compiler@6.1.2
+-- @angular/compiler-cli@6.1.2
+-- @angular/core@6.1.2
+-- @angular/forms@6.1.2
+-- @angular/http@6.1.2
+-- @angular/language-service@6.1.2
+-- @angular/platform-browser@6.1.2
+-- @angular/platform-browser-dynamic@6.1.2
+-- @angular/router@6.1.2
+-- @progress/kendo-angular-buttons@4.1.3
+-- @progress/kendo-angular-dateinputs@3.4.4
+-- @progress/kendo-angular-dropdowns@3.0.2
+-- @progress/kendo-angular-excel-export@2.1.1
+-- @progress/kendo-angular-grid@3.6.1
+-- @progress/kendo-angular-inputs@3.2.0
+-- @progress/kendo-angular-intl@1.4.1
+-- @progress/kendo-angular-l10n@1.2.0
+-- @progress/kendo-angular-popup@2.4.1
+-- @progress/kendo-data-query@1.4.1
+-- @progress/kendo-drawing@1.5.7
+-- @progress/kendo-theme-default@2.54.1
+-- @types/jasmine@2.8.8
+-- @types/jasminewd2@2.0.3
+-- @types/node@8.9.5
+-- codelyzer@4.2.1
+-- core-js@2.5.7
+-- jasmine-core@2.99.1
+-- jasmine-spec-reporter@4.2.1
+-- karma@1.7.1
+-- karma-chrome-launcher@2.2.0
+-- karma-coverage-istanbul-reporter@2.0.1
+-- karma-jasmine@1.1.2
+-- karma-jasmine-html-reporter@0.2.2
+-- protractor@5.3.2
+-- rxjs@6.2.2
+-- rxjs-compat@6.2.2
+-- ts-node@5.0.1
+-- tslint@5.9.1
+-- typescript@2.7.2
`-- zone.js@0.8.26
```

Browser:

- Chrome (desktop) Version 68.0.3440.106 (Official Build) (64-bit)

System:
- TypeScript version: 2.8.4 
- Node version: 6.1.0 
- Platform: Windows

**Comment by danielkaradachki** [comment]
I guess the expected behavior depends on how are you creating the dates received from the server. If they are not converted to local time and the user sees the same dates that are on the server then the current behavior is the correct one. However if the dates are converted to local time then they should be converted back to UTC when sending the filter. Both scenarios are valid so we will expose an option that specifies how should the filter dates be serialized. For now I can suggest to replace the dates in the resulting query string:
https://stackblitz.com/edit/angular-tchmjm?file=app/app.component.ts

**Comment by danielkaradachki** [comment]
Available in the latest [dev](https://www.telerik.com/kendo-angular-ui-develop/components/installation/development-builds/) version:
https://www.telerik.com/kendo-angular-ui-develop/components/dataquery/api/ODataSettings/

**Comment by danielkaradachki** [comment]
Available in 1.5.0
